src/BladeDirectivesServiceProvider.php

```diff
@@ -32,6 +28,14 @@ public function register()
             return "<?php try { echo \$__env->first({$expression}, \Illuminate\Support\Arr::except(get_defined_vars(), ['__data', '__path']))->render(); } catch (\InvalidArgumentException \$e) { if (!app()->environment('production')) { echo '<hr>'.__('View not found: :view', ['view' => implode(', ', [{$expression}][0])]).'<hr>'; } } ?>";
         });
 
+        Blade::directive('markdown', function ($markdown) {
+            return "<?php echo Str::markdown(e($markdown)); ?>";
```

Comment:
Do we want that escaped? As mixing html with markdown is usually allowed, escaping breaks that.

Reply:
Seems you're right in that it's allowed and even encouraged. I've removed the escaping and also added a null coalesce for safety.

Reply:
Supporting html would be nice, i do think we would want to read through 
https://github.com/thephpleague/commonmark/blob/dd8f9fb5fa5da12ce31e307e659ed90c2157b770/docs/2.0/security.md
though.
As currently this basically implements **no** security

Reply:
I've changed it now so that it can now be used with a 2nd parameter like

```blade
@markdown($text, 'strip')
```

where it defaults to `escape` if you don't supply it.